app/user/rest.py

```diff
@@ -142,41 +136,75 @@ def verify_user_code(user_id):
         increment_failed_login_count(user_to_verify)
         raise InvalidRequest("Code has expired", status_code=400)
 
-    if code_type == 'sms':
-        user_to_verify.current_session_id = str(uuid.uuid4())
-        user_to_verify.logged_in_at = datetime.utcnow()
-        user_to_verify.failed_login_count = 0
-        save_model_user(user_to_verify)
+    user_to_verify.current_session_id = str(uuid.uuid4())
+    user_to_verify.logged_in_at = datetime.utcnow()
+    user_to_verify.failed_login_count = 0
+    save_model_user(user_to_verify)
 
     use_user_code(code.id)
     return jsonify({}), 204
 
 
 @user_blueprint.route('/<uuid:user_id>/sms-code', methods=['POST'])
 def send_user_sms_code(user_id):
+    data = request.get_json()
+    user_to_send_to = validate_2fa_call(user_id, data, post_send_user_sms_code_schema)
+    if not user_to_send_to:
+        return jsonify({}), 204
+
+    secret_code = create_secret_code()
+    create_user_code(user_to_send_to, secret_code, SMS_TYPE)
+
+    mobile = data.get('to') or user_to_send_to.mobile_number
+    template = dao_get_template_by_id(current_app.config['SMS_CODE_TEMPLATE_ID'])
+
+    personalisation = {'verify_code': secret_code}
+
+    create_2fa_code(template, mobile, personalisation)
+    return jsonify({}), 204
+
+
+@user_blueprint.route('/<uuid:user_id>/email-code', methods=['POST'])
+def send_user_email_code(user_id):
```

Comment:
This is pretty much the same as the code for SMS, you should consolidate this method and pass in the template and the personalisation